### PR TITLE
esp32c3: Add initial support to esp32c3-devkit-rust-1 board

### PR DESCRIPTION
**Pull request by acassis**
## Summary
Add initial support to esp32c3-devkit-rust-1 board
## Impact
Users could use the esp32c3-devkit-rust-1 board with NuttX (and eventually run Rust on it over NuttX)
## Testing
esp32c3-devkit-rust-1

### CODE REVIEW

File: boards/risc-v/esp32c3/esp32c3-devkit-rust-1/src/esp32c3_autoleds.c [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+      set_led(true);
+      break;
+
+    case LED_PANIC:
+
+      /* For panic state, the LED is blinking */
+
+      set_led(true);
+      break;
```
Comment (pkarashchenko) [inline boards/risc-v/esp32c3/esp32c3-devkit-rust-1/src/esp32c3_autoleds.c:93]: please add 2 spaces for case block
also please add `default`case

**Review by pkarashchenko — COMMENTED** [review]
In general looks good.

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@acassis do you plan to mainline the rust porting to nuttx?

File: boards/risc-v/esp32c3/esp32c3-devkit-rust-1/src/esp32c3_autoleds.c [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+void board_autoled_off(int led)
+{
+  switch (led)
+    {
+    case LED_PANIC:
+
+      /* For panic state, the LED is blinking */
+
+      set_led(false);
+      break;
```
[the remark this answers is not in the document]
Comment (acassis) [reply boards/risc-v/esp32c3/esp32c3-devkit-rust-1/src/esp32c3_autoleds.c]: Thank you very much @pkarashchenko ! Done!

**Comment by acassis** [comment]
> @acassis do you plan to mainline the rust porting to nuttx?

Hi @xiaoxiang781216, this board was developed by a colleague that is Rust fan and it will be used in the Ferrous Systems training: https://ferrous-systems.com/go/trainings/esp32-c3-devkit-rust-1/ Although its initial purpose was to be used in that training, this is a nice board to be used with NuttX. About the Rust porting we need to find a way to better support "cargo" in our Building System, this is the next step to get Rustix compiled on NuttX.

**Comment by xiaoxiang781216** [comment]
> > @acassis do you plan to mainline the rust porting to nuttx?
> 
> Hi @xiaoxiang781216, this board was developed by a colleague that is Rust fan and it will be used in the Ferrous Systems training: https://ferrous-systems.com/go/trainings/esp32-c3-devkit-rust-1/ Although its initial purpose was to be used in that training, this is a nice board to be used with NuttX. About the Rust porting we need to find a way to better support "cargo" in our Building System,

How about to leverage CMake?

> this is the next step to get Rustix compiled on NuttX.

We are very interesting to support rust on NuttX.

**Comment by acassis** [comment]
> > > @acassis do you plan to mainline the rust porting to nuttx?
> > 
> > 
> > Hi @xiaoxiang781216, this board was developed by a colleague that is Rust fan and it will be used in the Ferrous Systems training: https://ferrous-systems.com/go/trainings/esp32-c3-devkit-rust-1/ Although its initial purpose was to be used in that training, this is a nice board to be used with NuttX. About the Rust porting we need to find a way to better support "cargo" in our Building System,
> 
> How about to leverage CMake?
> 
I think it is better to get CMake integrated and stable on NuttX first.

> > this is the next step to get Rustix compiled on NuttX.
> 
> We are very interesting to support rust on NuttX.

I think we can discuss it with @lupyuen and @sunfishcode how to do it. I think we need to create same generic rule in the make build system to run cargo when it enter inside a Rust directory, same way we do it automatically for regular C programs in the apps/ directory.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**